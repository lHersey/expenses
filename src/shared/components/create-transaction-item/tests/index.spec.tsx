import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import Providers from 'shared/providers/test-providers';

import CreateTransactionItem, { CREATE_TRANSACTION_ITEM_TEST_ID } from '../index';

describe('<CreateTransactionItem />', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
      expect.hasAssertions();
      const rendered = render(
        <Providers>
          <CreateTransactionItem label="Label" onPress={jest.fn()} icon="add-sharp" />
        </Providers>,
      ).toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('unit', () => {
    it('should render the text', () => {
      expect.hasAssertions();

      const label = 'My label :)';

      const { queryByText } = render(
        <Providers>
          <CreateTransactionItem label={label} onPress={jest.fn()} icon="add-sharp" />
        </Providers>,
      );

      expect(queryByText(label)).toBeTruthy();
    });

    it('should handle onPress', () => {
      expect.hasAssertions();

      const handleOnPressMock = jest.fn();
      const { getByTestId } = render(
        <Providers>
          <CreateTransactionItem label="label" icon="add-sharp" onPress={handleOnPressMock} />
        </Providers>,
      );

      const pressable = getByTestId(CREATE_TRANSACTION_ITEM_TEST_ID);
      fireEvent.press(pressable);

      expect(handleOnPressMock).toHaveBeenCalledWith();
    });
  });
});
