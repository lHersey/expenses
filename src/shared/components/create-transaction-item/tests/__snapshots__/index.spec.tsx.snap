// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTransactionItem /> snapshots should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 10,
          "flexDirection": "row",
          "marginTop": 20,
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
          "width": "100%",
        },
      ]
    }
    testID="CREATE_TRANSACTION_ITEM_TEST_ID"
  >
    <Text />
    <Text
      style={
        Array [
          Object {
            "color": "#5e5e5e",
            "fontFamily": "Inter_500",
            "fontSize": 16,
          },
          Object {
            "color": "#b0b0b0",
            "fontFamily": "Inter_700",
            "marginLeft": 10,
          },
        ]
      }
    >
      Label
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
